Code/Tools/FBuild/FBuildCore/FBuild.cpp

```diff
@@ -237,6 +237,11 @@ bool FBuild::GetTargets( const Array< AString > & targets, Dependencies & outDep
 {
     ASSERT( !targets.IsEmpty() );
 
+    if ( m_DependencyGraph == nullptr )
+    {
+        return false;
+    }
```

Comment:
It's not valid to call this function (or any other function that derefs the m_DependencyGraph) without first successfully initializing the FBuild class (which creates the m_DependencyGraph. I think none of these nullptr guards make sense.

Was there a particular crash you saw?

Reply:
Numerous tests in the FBuildTest.exe initialize an FBuild object but don't pass in a const char * nodeGraphDBFile. Since there is no db file, m_DependencyGraph is null, resulting in a crash later in FBuild Initialize(). For these cases, FBuild Initialize() needs to return before it tries to access the graph.
Example crash stack:

>	FBuildTest.exe!NodeGraph::FindNodeInternal(const AString & fullPath) Line 1269	C++
 	FBuildTest.exe!NodeGraph::FindNode(const AString & nodeName) Line 613	C++
 	FBuildTest.exe!FBuild::Initialize(const char * nodeGraphDBFile) Line 171	C++
 	FBuildTest.exe!TestAlias::ReflectionAliasResolution() Line 62	C++
 	FBuildTest.exe!TestAlias::RunTests() Line 31	C++
 	FBuildTest.exe!UnitTestManager::RunTests(const char * testGroup) Line 128	C++
 	FBuildTest.exe!main(int __formal, char * * __formal) Line 51	C++
 	[External Code]

Reply:
nodeGraphDBFile is allowed to be null because it overrides the default.

Is it possible you have other local changes that cause this? All the tests are currently passing.

Reply:
This crash happens when nodeGraphDBFile is null, then FBuild Initialize() parses the .bff file from m_Options.m_ConfigFile. But in many of the tests, the bff file contains errors (to test error cases), yielding a null m_DependencyGraph and the resulting crash.

Reply:
Do you have a specific example of where you are seeing this problem? All the tests that expect failure during parsing should not be calling any functions that require the DependencyGraph post failed Initialize. It is not valid to call any on functions that interact with the dependency graph if Initialize() has failed, so silently returning would be covering up a problem and is not something we want to do.

As I say, the tests are currently all passing, so I suspect any problems are more likely to do with local changes you have, perhaps violating the expected usage pattern.

Reply:
Thanks. You're right, I pulled out these changes and all seems okay. Disregard?